pkg/controller/replicaset/replica_set.go

```diff
@@ -59,9 +58,8 @@ const (
 	statusUpdateRetries = 1
 )
 
-func getRSKind() schema.GroupVersionKind {
-	return v1beta1.SchemeGroupVersion.WithKind("ReplicaSet")
-}
+// controllerKind contains the schema.GroupVersionKind for this controller type.
+var controllerKind = v1beta1.SchemeGroupVersion.WithKind("ReplicaSet")
```

Comment:
There is unfortunately a lot of similar-but-not-yet-shareable code between all the controllers, and I wanted to introduce a convention of having `controllerKind` in the package so I can copy code between controllers and it would understand how to ask "what is my kind?".

I figured it serves a similar purpose to the `SchemeGroupVersion` variable itself:

https://github.com/kubernetes/kubernetes/blob/dd22743b54f280f41e68f206449a13ca949aca4e/pkg/apis/extensions/v1beta1/register.go#L29

In that case, not only is the read-only nature merely a convention, but the variable is package public. So I used it as precedent that read-only-by-convention variables are ok. If that's the case, then making it a function seems like unnecessary overhead both cognitively and at runtime.